### DISCUSSION TITLE
Handling Arrays in Query Parameters with Nuqs Library

### DISCUSSION THREAD

**Discussion by SergGrey1992** [47ng/nuqs#517] [Q&A] [answered]
Hello,

I'm using the Nuqs library in my Next.js 14.1.0 project and I'm seeking guidance on handling arrays within URL query parameters. Specifically, I'd like to know:

1) Adding Items: How can I add items to an array in the query parameters? Is there a specific method or best practice for doing this using Nuqs?

2) Deleting Items: What is the recommended way to remove items from an array in the query parameters with Nuqs?

3) Reading Query on Server: How can I read and process these query parameters on the server side? Are there any specific considerations or methods in Nuqs for server-side handling?

Any examples or documentation references on these points would be greatly appreciated.

I couldn't find the necessary information in the documentation, so I'm asking here. I really liked your library!!))

**Comment by franky47** [comment, marked as answer]
If you're talking about multiple keys in the URL giving an array output like URLSearchParam does, this is not currently supported, discussion about API ideas is here: https://github.com/47ng/nuqs/discussions/484

There is however a `parseAsArrayOf` parser that lets you put an array under a single search param key: https://nuqs.47ng.com/docs/parsers#arrays

As for best practices with setting state arrays, it's the same as for a `useState`: no mutation of the array (so no push/pop), always return a new instance with the added item or filter out the one to remove.
